features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.types.listing/listing.js

```diff
@@ -32,30 +29,52 @@ function onRequest(context) {
 
     if (data.data) {
         var deviceTypes = data.data;
-        var deviceTypesList = [];
+        var deviceTypesList = [], virtualDeviceTypesList = [];
         for (var i = 0; i < deviceTypes.length; i++) {
 
             var deviceTypeLabel = deviceTypes[i].name;
             var configs = utility.getDeviceTypeConfig(deviceTypeLabel);
-
-            if (configs && configs[DTYPE_CONF_DEVICE_TYPE_KEY][DTYPE_CONF_DEVICE_TYPE_LABEL_KEY]) {
-                deviceTypeLabel = configs[DTYPE_CONF_DEVICE_TYPE_KEY][DTYPE_CONF_DEVICE_TYPE_LABEL_KEY];
+            var deviceCategory = "device";
+            if (configs) {
+                if (configs[DTYPE_CONF_DEVICE_TYPE_KEY][DTYPE_CONF_DEVICE_CATEGORY]) {
+                    deviceCategory = configs[DTYPE_CONF_DEVICE_TYPE_KEY][DTYPE_CONF_DEVICE_CATEGORY];
+                }
+                if (configs[DTYPE_CONF_DEVICE_TYPE_KEY][DTYPE_CONF_DEVICE_TYPE_LABEL_KEY]) {
+                    deviceTypeLabel = configs[DTYPE_CONF_DEVICE_TYPE_KEY][DTYPE_CONF_DEVICE_TYPE_LABEL_KEY];
+                }
             }
             deviceTypesList.push({
                 "hasCustTemplate": false,
                 "deviceTypeLabel": deviceTypeLabel,
                 "deviceTypeName": deviceTypes[i].name,
-                "deviceCategory": configs[DTYPE_CONF_DEVICE_TYPE_KEY][DTYPE_CONF_DEVICE_CATEGORY],
+                "deviceCategory": deviceCategory,
                 "deviceTypeId": deviceTypes[i].id
             });
-
+            //if (deviceCategory == 'virtual'){
```

Comment:
Acknowledged and fixed with commit https://github.com/wso2/carbon-device-mgt/commit/0133cddc683fe7bf2c381a3e338d1c027f6ff1b0